### ISSUE TITLE
[FEAT] Implement BinaryEvaluator (KS, AUC, Gini, Brier, log-loss)

### ISSUE THREAD

**Issue by Cristiano2132** [Cristiano2132/model-track-cr#43]
## 📋 Context

The `evaluation/` module exists but is completely empty. This is the most critical gap: the library declares itself as a "model monitoring toolkit" but has no model evaluation capabilities.

`BinaryEvaluator` is the foundation — it covers the binary classification use case already demonstrated in the notebook (`ieee_fraud_eda_baseline.ipynb`), where KS and AUC are computed manually inline.

## 🎯 Class Design

```mermaid
classDiagram
    class BaseEvaluator {
        <<abstract>>
        +task_type: TaskType
        +evaluate(y_true, y_pred, y_proba) dict
        +report(df, target, score_col) DataFrame
    }

    class BinaryEvaluator {
        +task_type = TaskType.BINARY
        +positive_class: int = 1
        +evaluate(y_true, y_proba) dict
        +report(df, target, score_col) DataFrame
        -_compute_ks(y_true, y_proba) float
        -_compute_gini(auc) float
        -_compute_brier(y_true, y_proba) float
    }

    BaseEvaluator <|-- BinaryEvaluator
```

### Metrics returned by `evaluate()`

| Metric | Description |
|--------|-------------|
| `auc` | ROC AUC (sklearn) |
| `ks` | Kolmogorov-Smirnov statistic |
| `gini` | Gini coefficient = 2*AUC - 1 |
| `brier_score` | Brier score (calibration quality) |
| `log_loss` | Cross-entropy loss |

### Usage

```python
from model_track.evaluation import BinaryEvaluator

evaluator = BinaryEvaluator()
metrics = evaluator.evaluate(y_true=df["target"], y_proba=df["pred_proba"])
# {"auc": 0.87, "ks": 0.52, "gini": 0.74, "brier_score": 0.08, "log_loss": 0.23}

report = evaluator.report(df, target="target", score_col="pred_proba")
# DataFrame with metrics per period (groupby time column)
```

### `report()` — temporal metrics

Accepts an optional `date_col` argument to compute metrics per time period:

```python
report = evaluator.report(
    df, target="target", score_col="pred_proba", date_col="tm_yymm_str"
)
# Returns DataFrame: period | auc | ks | gini | brier_score
```

## ✅ Acceptance Criteria

- [x] `BinaryEvaluator` implements all 5 metrics
- [x] `report()` supports per-period aggregation via `date_col`
- [x] `BaseEvaluator` ABC defined in `evaluation/base.py`
- [x] `BinaryEvaluator` exported from `model_track.evaluation.__init__`
- [x] Unit tests covering: perfect model, random model, edge cases (all same class)
- [x] Statistical tests (Hypothesis) for metric boundary conditions
- [x] `mypy --strict` passes

## 📁 New Files

- `src/model_track/evaluation/base.py`
- `src/model_track/evaluation/binary.py`
- `src/model_track/evaluation/__init__.py` (update)
- `tests/unit/evaluation/test_binary_evaluator.py`
- `tests/statistical/evaluation/test_binary_properties.py`

---
**Observações Finais**: A implementação foi realizada com sucesso! A base abstrata `BaseEvaluator` já prevê integração padronizada com outras classes, o `report()` suporta dados temporais com skips graciosos quando há apenas uma classe e os testes alcançam limites rigorosos através do `hypothesis`. O código passa 100% no `mypy --strict` e atinge 100% de coverage na rotina unitária e de propriedades.

**Comment by Cristiano2132** [comment]
🚀 Starting implementation of `BinaryEvaluator`. Branch: `feature/43-binary-evaluator`

**Plan:**
- `BaseEvaluator` ABC in `evaluation/base.py`
- `BinaryEvaluator` with KS, AUC, Gini, Brier, log-loss in `evaluation/binary.py`
- Temporal `report()` via `date_col`
- Unit + statistical (Hypothesis) tests
